File: challenges/007/camel_case.py
def to_camel_case(text):
    words = text.replace("_", "-").split("-")
    converted_words = []

    converted_words.append(words[0])
    for word in words[1:]:
        converted_words.append(word[0].upper() + word[1:])

    return ''.join(converted_words)


# More pythonic version
def to_camel_case(text):
    words = text.replace("_", "-").split("-")
    return words[0] + ''.join(word.capitalize() for word in words[1:])

File: challenges/007/camelCase.js
function toCamelCase(str){
    const words = str.split(/[-_]/);
    const convertedWords = [];

    convertedWords.push(words[0])

    for (const word of words.slice(1)) {
        let convertedWord = word.charAt(0).toUpperCase() + word.slice(1);
        convertedWords.push(convertedWord)
    }

    return convertedWords.join('')
}


// Simpler version using slice and map
function toCamelCase(str) {
    return str
        .split(/[-_]/)
        .map((word, i) => i ? word[0].toUpperCase() + word.slice(1) : word)
        .join('');
}
